src/realm/array.cpp

```diff
@@ -1717,9 +1717,9 @@ void Array::alloc(size_t init_size, size_t width)
             // field in the header.
             size_t new_capacity_bytes = capacity_bytes * 2;
             if (new_capacity_bytes < capacity_bytes) // overflow detected, clamp to max
```

Comment:
Oh, now I understand. @jedelbo would it be sufficient to replace the check with a comment and an assert? Then we at least show that overflow has been thought about.